scripts/launch-xrd

```diff
@@ -134,10 +134,18 @@ _runcmd() {
 # Pulls an image if not loaded already. Return code 1 if image is unavilable
 #   Arg 1: image name
 _pull_image() {
-    if ! "$CONTAINER_EXE" pull "$1" || ! "$CONTAINER_EXE" inspect "$1" &>/dev/null; then
-        echo "Unrecognised image '$1' - should be either an image loaded " \
-            "in the local $CONTAINER_MGR registry or an image that can be pulled." >&2
+    if [[ $DRY_RUN == 1 ]]; then
```

Comment:
I probably would've put this on line 308 rather than moving that echo to here. That way it seems clearer that we only try to pull if not doing a dry-run.

Reply:
I thought the code would look uglier with 3 nested ifs. But fair enough